### DISCUSSION TITLE
单机部署多套环境无法正常启动

### DISCUSSION THREAD

**Discussion by jiangtao007** [apolloconfig/apollo#4085] [Q&A]
大家好，我在单机上部署2套环境的时候出现一个服务无法正常启动的问题，数据库已经单独出来了，然后adservice和configservice也是2个独立的服务，修改了其中一个服务的star.sh脚本里面的SERVER_PORT=${SERVER_PORT:=18080}，但是这个服务怎么都无法启动，也看到不到日志，最后很长时间后直接说启动失败，请问你们有遇到过这种情况吗？

**Comment by nobodyiam** [comment]
看下日志目录是否没有写权限 /opt/logs，或者到 /tmp 目录下看看是否有日志

**Reply by jiangtao007** [reply]
多谢回复，我修改了自定义的日志目录，我是用root用户启动的。权限问题应该也没有，然后/tmp下面有目录，但是里面没有任何日志，我今天再尝试将目录权限改成777试一下。

**Reply by nobodyiam** [reply]
如果仍然失败的话，可以执行`export APOLLO_RUN_MODE=Docker`改为前台启动，然后再执行`startup.sh`

**Reply by jiangtao007** [reply]
终于看到报错了，报的是端口冲突了，但是我项目下面已经改了8080了，有点奇怪。
Description:

Web server failed to start. Port 8080 was already in use.

Action:

Identify and stop the process that's listening on port 8080 or configure this application to listen on another port.
但是我在向项目目录下去查这个端口发现没有8080的配置，我已经改成了8081
SERVER_PORT=${SERVER_PORT:=8081}
请问一下这个是什么问题呀？感觉像是读错了配置文件或者我改的配置文件没有生效

**Reply by nobodyiam** [reply]
> SERVER_PORT=${SERVER_PORT:=8081}

看看是否环境变量定义了 SERVER_PORT

**Reply by jiangtao007** [reply]
没有，我是下载编译好的包是不是需要自己改配置再编译一遍？

**Reply by nobodyiam** [reply]
只要改 startup.sh 就可以了
